go/enclave/db/storage.go

```diff
@@ -213,15 +215,17 @@ func (s *storageImpl) Proof(r *core.Rollup) *types.Block {
 	return v
 }
 
-func (s *storageImpl) FetchBlockState(hash common.L1RootHash) (*core.BlockState, bool) {
+func (s *storageImpl) FetchBlockState(hash common.L1RootHash) (*core.BlockState, map[uuid.UUID][]*types.Log, bool) {
 	bs := obscurorawdb.ReadBlockState(s.db, hash)
-	if bs != nil {
-		return bs, true
+	logs := obscurorawdb.ReadBlockLogs(s.db, hash)
```

Comment:
I guess at some point this thing would pop the memory, if it can grown unbounded in two dimensions (number of subscriptions open and log accumulation over time).

I guess we'll just have to put a reasonable cap on number of open subscriptions, and have a chat with Tudor about how many rollups back we keep logs for.